lua/paramset.lua

```diff
@@ -43,40 +43,41 @@ function ParamSet:add(args)
   local param = args.param -- param is mandatory
   table.insert(self.params, param)
   self.count = self.count + 1
-  self.lookup[param.name] = self.count
+  self.lookup[param.id] = self.count
   if args.action then -- action is optional
     param.action = args.action
   end
 end
 
 --- add number
-function ParamSet:add_number(name, min, max, default)
-  self:add { param=number.new(name, min, max, default) }
+function ParamSet:add_number(id, name, min, max, default)
+  self:add { param=number.new(id, name, min, max, default) }
 end
 
 --- add option
-function ParamSet:add_option(name, options, default)
-  self:add { param=option.new(name, options, default) }
+function ParamSet:add_option(id, name, options, default)
+  p = option.new(id, name, options, default)
+  self:add { param=option.new(id, name, options, default) }
 end
 
 --- add control
-function ParamSet:add_control(name, controlspec, formatter)
-  self:add { param=control.new(name, controlspec, formatter) }
+function ParamSet:add_control(id, name, controlspec, formatter)
+  self:add { param=control.new(id, name, controlspec, formatter) }
```

Comment:
alternative would be `self:add { id=id, param=control.new(name, controlspec, formatter)`

Reply:
also agreed, but i'd rather make this change in a different PR.